community/index/src/main/java/org/neo4j/index/internal/gbptree/GroupingRecoveryCleanupWorkCollector.java

```diff
@@ -48,47 +50,64 @@ public GroupingRecoveryCleanupWorkCollector( JobScheduler jobScheduler )
     @Override
     public void init()
     {
+        started = false;
         jobs.clear();
     }
 
     @Override
     public void add( CleanupJob job )
     {
+        if ( started )
+        {
+            throw new IllegalStateException( "Index clean jobs can't be added after collector start." );
+        }
         jobs.add( job );
     }
 
     @Override
     public void start()
     {
-        jobScheduler.schedule( recoveryCleanup, allJobs() );
-    }
-
-    @Override
-    public void stop()
-    {   // no-op
+        scheduleJobs();
+        started = true;
     }
 
-    @Override
-    public void shutdown()
-    {   // no-op
+    private void scheduleJobs()
+    {
+        jobScheduler.schedule( recoveryCleanup, allJobs() );
     }
 
     private Runnable allJobs()
     {
         return () ->
         {
             CleanupJob job;
+            Exception jobsException = null;
             while ( (job = jobs.poll()) != null )
             {
                 try
                 {
                     job.run();
                 }
+                catch ( Exception e )
+                {
+                    if ( jobsException == null )
```

Comment:
Not important, but there's the [Exceptions#withSuppressed](https://github.com/neo4j/neo4j/blob/3.4/community/common/src/main/java/org/neo4j/helpers/Exceptions.java#L141) utility that could be used here.